lib/optimove.dart

```diff
@@ -0,0 +1,255 @@
+    Map<String, dynamic> result = Map<String, dynamic>.from(await _methodChannel.invokeMethod('inAppGetInboxSummary'));
+
+    return OptimoveInAppInboxSummary(result['totalCount'], result['unreadCount']);
+  }
+
+  static Future<void> updateConsentForUser(bool consentGiven) async {
+    return _methodChannel.invokeMethod('inAppUpdateConsent', <String, bool>{
+      'consentGiven': consentGiven
+    });
+  }
+
+  static Future<List<OptimoveInAppInboxItem>> getInboxItems() async {
+    var data = await _methodChannel.invokeMethod('inAppGetInboxItems');
+
+    if (data == null) {
+      return [];
+    }
+
+    var items = List.from(data).map((e) => OptimoveInAppInboxItem.fromMap(Map<String, dynamic>.from(e))).toList();
+    return items;
+  }
+
+  static Future<OptimoveInAppPresentationResult> presentInboxMessage(OptimoveInAppInboxItem item) async {
+    var result = await _methodChannel.invokeMethod<int>('inAppPresentInboxMessage', <String, int>{
+      'id': item.id
+    });
+
+    return result != null ? OptimoveInAppPresentationResult.values[result] : OptimoveInAppPresentationResult.Failed;
+  }
+
+  static Future<bool> deleteMessageFromInbox(OptimoveInAppInboxItem item) async {
+    var result = await _methodChannel.invokeMethod<bool>('inAppDeleteMessageFromInbox', <String, int>{
+      'id': item.id
+    });
+
+    return result ?? false;
+  }
+
+  static Future<bool> markAsRead(OptimoveInAppInboxItem item) async {
+    var result = await _methodChannel.invokeMethod<bool>('inAppMarkAsRead', <String, int>{
+      'id': item.id
+    });
+
+    return result ?? false;
+  }
+
+  static void setPushReceivedHandler(void Function(OptimovePushNotification)? pushReceivedHandler) {
+    _pushReceivedHandler = pushReceivedHandler;
+    initStreamIfNeeded();
+  }
+
+  static void setPushOpenedHandler(void Function(OptimovePushNotification)? pushOpenedHandler) {
+    _pushOpenedHandler = pushOpenedHandler;
+    initStreamIfNeeded();
+  }
+
+  static void setDeeplinkHandler(void Function(OptimoveDeepLinkOutcome)? deepLinkHandler) {
+    _deepLinkHandler = deepLinkHandler;
+    initStreamIfNeeded();
+  }
+
+  static void setInAppDeeplinkHandler(void Function(Map<String, dynamic>)? inAppDeepLinkHandler) {
+    _inAppDeepLinkHandler = inAppDeepLinkHandler;
+    initStreamIfNeeded();
+  }
+
+  static void setOnInboxUpdatedHandler(Function? handler) {
+    _inboxUpdatedHandler = handler;
+    initStreamIfNeeded();
+  }
+
+  static void initStreamIfNeeded() {
+    if (!listenersExist()) {
+      _eventStream?.cancel();
+      _eventStream = null;
+      return;
+    }
+
+    if (_eventStream != null) {
+      return;
+    }
+
+    initStream();
+  }
+
+  static bool listenersExist() {
+    return _pushOpenedHandler != null || _pushReceivedHandler != null || _deepLinkHandler != null || _inboxUpdatedHandler != null || _inAppDeepLinkHandler != null;
+  }
+
+  static void initStream() {
+    _eventStream = _eventChannel.receiveBroadcastStream().listen((event) {
```

Comment:
Are there any potential issues with using one stream for all event types?

For instance, if we have a queued DDL or push open event, but the handler for (say) in-app is registered first, we'd flush the queued events when there are no listeners waiting on the dart app.

Reply:
Yes you are right, for the queued events this might be a problem as the listeners are set after the events occur.
I would like to keep the flexibility to register listeners in different parts, thinking about splitting the channels to delayed and immediate or something.

Reply:
I agree it's good to keep the ability to set the listeners separate, so sounds good.

I wonder if splitting into two channels is enough? i.e. can a push open event be swallowed by registration of the DDL handler (for example). Sequence:

1. tap notification (cold start)
2. send open on delayed sink
3. dart app sets DDL handler
4. delayed queue flushes
5. dart app sets push handler
6. nothing queued left

Perhaps I'm misunderstanding the split, let's discuss in standup.

Reply:
The delayed listeners are set together now